appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/processor/ApplicationProcessor.java

```diff
@@ -150,6 +154,26 @@
      */
     public ApplicationProcessor(Set<Class<?>> appClasses) {
         this.classes = appClasses;
+        this.classes.removeIf(cls -> cls.isInterface() || Modifier.isAbstract(cls.getModifiers()));
+        if (classes != null) {
+            addInnerClasses();
+        }
+    }
+
+    private void addInnerClasses() {
+        List<Class<?>> topLevelClasses = new ArrayList<>(classes);
+        for (Class<?> topLevelClass : topLevelClasses) {
+            addInnerClasses(topLevelClass);
+        }
+    }
+
+    private void addInnerClasses(Class<?> topLevelClass) {
```

Comment:
I don't think the application processor cares about this. Is it possible to put this in the walker?

Reply:
I agree that this should be in the walker. Moving it there does fail tests though. Reason is that the list of classes is used for the resource mapping which is created before the walker would add inner classes. To me this also looks like it never should have been in the processor but be purely within the walker. The processor has the `addParameter` method which also creates a context wherefore it also recreates a resource mapping again (each time). I think this is kind of a work around. The walking logic of how to process fields annotated with a parameter annotation should be in the walker which would avoid creating the resource mapping many times over and would allow moving the creation of the mapping into the walker which allows the walker to first add the inner classes so these are also considered in the resource mapping.

While this seems a clear resolution to the issue to me this is a more complicated refactoring disentangling processor and walker so the result is still the same. I am note sure we should do this in the context of this issue.